### PR TITLE
Performance optimizations

### PR DESCRIPTION
**Pull request by maschnetwork**
**Description of changes:**

- Micronaut: Simplified GraalVM native image building with built-in support
- Micronaut: Refactored from controller to handler methods as both Quarkus and Spring example are using a functional approach rather than a controller approach. 
- Micronaut: Used URLHttpConnection instead of default client in Micronaut sample, as both Spring & Quarkus were using it as well
- Quarkus & Spring: Added Tiered compilation to Quarkus & Spring JVM applications as Micronaut example was using Tiered as well

New benchmark results (Only tested with 50 concurrent requests / s for 60 seconds - which should give a first impression):

**Micronaut JVM**:
micronaut-jvm

Current p50 cold start: *8505* vs. new p50 cold start: *4835* = ***+45%  Improvement***

**Quarkus JVM**:
quarkus-jvm

Current p50 cold start: *6384* vs. new p50 cold start: *3755* = ***+42% Improvement***

**Spring JVM:**
spring-jvm

Current p50 cold start: *12673* vs. new p50 cold start: *5370* = ***+58% Improvement***

**Micronaut Native:**
micronaut-native

ps: Didn't add other native results as they should be similar as there were no changes.

*Small caveat*:
- Had to add reflect config to Micronaut sample since the handler is loaded dynamically via the Env-Var + logging classes from the AWS SDK. 

*Ask for input*:
- I would appreciate if someone would contribute a full load test to this example and add the updated numbers (Maybe @dmahapatro or @muraria ). Both for double-checking as well as filtering out variance.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: micronaut/README.md [hunk 1]
```diff
@@ -74,10 +59,10 @@ Using this CloudWatch Logs Insights query you can analyse the latency of the req
 The query separates cold starts from other requests and then gives you p50, p90 and p99 percentiles.
 
 ```
-filter @type="REPORT"
+
+```filter @type="REPORT"
```
Comment (msailes) [inline micronaut/README.md:63]: Should this be on a new line?

File: micronaut/README.md [hunk 2]
```diff
@@ -15,23 +15,8 @@ Make sure the app name used here matches with the `STACK_NAME` present under `lo
...
-```
-Make sure to replace `MICRONAUT_DIR_ABSOLUTE_PATH` with absolute path to micronaut directory.
- 
-Once docker downloads the image and runs, you would see a bash command that will run inside docker container.
-Run below command:
-
-```bash
-mvn clean package -Dpackaging=native-image -Pnative
-exit
+mvn clean package -Dpackaging=docker-native -Dmicronaut.runtime=lambda -Pgraalvm
```
Comment (msailes) [inline micronaut/README.md:19]: Great simplification!

File: micronaut/src/main/java/software/amazonaws/example/product/handler/CreateProductHandler.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import io.micronaut.core.annotation.Introspected;
+import io.micronaut.function.aws.MicronautRequestHandler;
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+import software.amazon.awssdk.http.HttpStatusCode;
+import software.amazon.awssdk.http.SdkHttpMethod;
+import software.amazonaws.example.product.dao.ProductDao;
+import software.amazonaws.example.product.entity.Product;
+
+
```
Comment (msailes) [inline micronaut/src/main/java/software/amazonaws/example/product/handler/CreateProductHandler.java:18]: nit: extra white space

File: quarkus/template.yaml [hunk 4]
```diff
@@ -14,6 +14,7 @@ Globals:
     MemorySize: 1024
     Environment:
       Variables:
+        JAVA_TOOL_OPTIONS: "-XX:+TieredCompilation -XX:TieredStopAtLevel=1"
```
Comment (muraria) [inline quarkus/template.yaml:17]: This is a welcome update. But, I don't think Micronaut is using tired compilation either. Can you do the same for Micronaut?

File: micronaut/README.md (hunk 2 above)
Comment (muraria) [reply micronaut/README.md:19]: Verified that it works on windows

File: micronaut/pom.xml [hunk 5]
```diff
@@ -92,7 +93,20 @@
...
+        </exclusion>
+        <exclusion>
+          <groupId>software.amazon.awssdk</groupId>
+          <artifactId>netty-nio-client</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
+    <dependency>
+      <groupId>software.amazon.awssdk</groupId>
+      <artifactId>url-connection-client</artifactId>
```
[the remark this answers is not in the document]
Comment (maschnetwork) [reply micronaut/pom.xml:109]: I wanted to keep it consistent with the other samples. Maybe we can add CRT as a separate PR.

File: micronaut/README.md (hunk 2 above)
Comment (muraria) [reply micronaut/README.md:19]: Sorry, spoke a little too soon. On windows, it needed a small change in the command line option. Still much simpler than before. This is what worked:
mvn clean package -Dpackaging=docker-native -Dmicronaut_runtime=lambda -Pgraalvm

Comment (muraria) [reply micronaut/README.md:19]: Sorry. Spoke a little too soon. Seems like there is a slight difference with cmd vs powershell in windows. Above command works with cmd. However, this is what worked in powershell:
`mvn clean package -Dpackaging=docker-native -Dmicronaut_runtime=lambda -Pgraalvm`
So, if underscore works on mac, we should change the above command.

**Review by muraria — APPROVED** [review]
Approving the PR. 
Going to create separate issues for AWS CRT and moving to latest version of Micronaut

**Pull request merged**